File: backend/app/api/diagnostics.py
"""Environment diagnostics API."""

from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.diagnostics import DiagnosticsResponse
from app.services import diagnostics_service

router = APIRouter(prefix="/diagnostics", tags=["diagnostics"])


@router.get("", response_model=DiagnosticsResponse)
def get_diagnostics(db: Session = Depends(get_db)) -> DiagnosticsResponse:
    return DiagnosticsResponse(**diagnostics_service.get_diagnostics(db))

File: frontend/src/api/diagnostics.ts
import { api } from "./client";
import type { DiagnosticsResponse } from "./types";

/** GET /api/diagnostics：设置、执行器、数据库与题库导入状态。 */
export async function getDiagnostics(): Promise<DiagnosticsResponse> {
  const { data } = await api.get<DiagnosticsResponse>("/diagnostics");
  return data;
}
